plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy

```diff
@@ -0,0 +1,266 @@
+    private static final double DEFAULT_RETRY_POLICY_JITTER = 0.5
+
+    // The HttpClient instance used to send requests
+    private final HttpClient httpClient = newDefaultHttpClient()
+
+    // The RetryPolicy instance used to retry requests
+    private final RetryPolicy retryPolicy = newDefaultRetryPolicy(SERVER_ERRORS)
+
+    // Nextflow session
+    private final Session session
+
+    // Platform endpoint to use for requests
+    private final String endpoint
+
+    // Platform access token to use for requests
+    private final String accessToken
+
+    /**
+     * Constructor for the class. It initializes the session, endpoint, and access token.
+     */
+    TowerFusionEnv() {
+        this.session = Global.session as Session
+        final towerConfig = session.config.navigate('tower') as Map ?: [:]
+        final env = SysEnv.get()
+        this.endpoint = endpoint0(towerConfig, env)
+        this.accessToken = accessToken0(towerConfig, env)
+    }
+
+    /**
+     * Return any environment variables relevant to Fusion execution. This method is called
+     * by {@link nextflow.fusion.FusionEnvProvider#getEnvironment} to determine which
+     * environment variables are needed for the current run.
+     *
+     * @param scheme The scheme for which the environment variables are needed (currently unused)
+     * @param config The Fusion configuration object
+     * @return A map of environment variables
+     */
+    @Override
+    Map<String, String> getEnvironment(String scheme, FusionConfig config) {
+
+        // TODO(amiranda): Hardcoded for now. We need to find out how to obtain
+        // the concrete product SKU and version. Candidate: FusionConfig?
+        final product = 'fusion'
+        final version = '2.4'
+
+        try {
+            final token = getLicenseToken(product, version)
+            return [
+                'FUSION_LICENSE_TOKEN': token,
+            ]
+        } catch (Exception e) {
+            log.warn("Error retrieving Fusion license information: ${e.message}")
+            return [:]
+        }
+    }
+
+    /**
+     * Send a request to Platform to obtain a license-scoped JWT for Fusion. The request is authenticated using the
+     * Platform access token provided in the configuration of the current session.
+     *
+     * @throws AbortOperationException if a Platform access token cannot be found
+     *
+     * @return The signed JWT token
+     */
+    protected String getLicenseToken(product, version) {
+        // FIXME(amiranda): Find out how to obtain the product and version
+        // Candidate: FusionConfig?
+
+        if (accessToken == null) {
+            throw new AbortOperationException("Missing personal access token -- Make sure there's a variable TOWER_ACCESS_TOKEN in your environment")
+        }
+
+        final req = HttpRequest.newBuilder()
+            .uri(URI.create("${endpoint}/${LICENSE_TOKEN_ENDPOINT}").normalize())
```

Comment:
`LICENSE_TOKEN_ENDPOINT` should not be a fully qualified URI

Reply:
I don't follow: `LICENSE_TOKEN_ENDPOINT` is set to `license/token/`, so what am I missing? Could you elaborate?

Reply:
an endpoint is usually a fully qualified URI, see [here](https://github.com/nextflow-io/nextflow/blob/6183fdf9dbd9114f5ee86d11bc5e69cbd06501a6/plugins/nf-wave/src/main/io/seqera/wave/plugin/config/TowerConfig.groovy#L53-L53) for example

This looks like more a path, always is the constant really needed?

Reply:
Ah, I get what you mean now. I can either rename it to something like `LICENSE_TOKEN_PATH` or just get rid of the constant altogether. I generally prefer to use constants in case we ever decide to change it in the future :shrug:.

Reply:
Addressed with 20c8a535dca9cb55e84f19682cfe3031baa75309